### DISCUSSION TITLE
How to use XML lsp?

### DISCUSSION THREAD

**Discussion by Navyashree0923** [emacs-lsp/lsp-mode#3527] [Q&A] [answered]
Hi, My goal is to activate XML language server using LSP mode. Unfortunately, I was unable to complete and I cannot see any suggestions or auto-completion but I was able to see the spell check, auto-completion and suggestions for Java. I have configured it by referring this doc - https://emacs-lsp.github.io/lsp-java/ and added 
```
(require 'lsp-java)
(add-hook 'java-mode-hook #'lsp)
```
 to `init.el` file

I tried by referring to this doc for XML  -  https://emacs-lsp.github.io/lsp-mode/page/lsp-xml/  and tried executing the command like   
M-x lsp-install-server RET xmlls RET and I can see the error as lsp server doesn't exist .
I also tried to add this content 
```
(require 'lsp-mode)
(add-hook 'xml-mode-hook #'lsp)
```
to `init.el` file 
Also tried to put the content 
```
(require 'lsp-xml)
(add-hook 'xml-mode-hook #'lsp)
```
to `init.el` file 
Any ideas or any suggestions how to make it working ?

**Comment by yyoncho** [comment]
What do do you get when you do `M-: major-mode RET`

**Reply by Navyashree0923** [reply]
I got -No match found

**Reply by yyoncho** [reply]
I don't think you can get that.

**Reply by yyoncho** [reply]
Most likely the major mode in your buffer is not xml-mode but either nxml-mode or web-mode. So the hook you are setting is not correct.

**Reply by Navyashree0923** [reply]
Sorry my bad, you're correct and the value is ‘nxml-mode’
```
Its value is ‘nxml-mode’
Original value was ‘fundamental-mode’
Local in buffer b.xml; global value is fundamental-mode
```

**Comment by yyoncho** [comment, marked as answer]
`(add-hook 'xml-mode-hook #'lsp)` that should be `(add-hook 'nxml-mode-hook #'lsp)`

**Reply by Navyashree0923** [reply]
Yes I made the changes in the init.el file and now I could see the xml related suggestions.
Thank you for the help
